### ISSUE TITLE
neo4j-admin import error by use docker image 4.4.29, but done by use 4.4.26 or 4.3.24 or 5.15.0

### ISSUE THREAD

**Issue by aicodex** [neo4j/neo4j#13373]
- Neo4j version: 4.4.27 - 4.4.29
- Operating system: Ubuntu20.04
API: Docker
- Expected behavior
- Actual behavior
- 
### Steps to reproduce
1. Pull the image: `docker pull neo4j/neo4j:4.4.27`
2. Create a directory on the host that will be a mount for the data: `mkdir /home/your_user/neo4j_data mkdir /home/your_user/import`
3. Start a new container that mounts to this directory: 

docker run --interactive --tty --rm -e LANG=zh_CN.utf8 -e LC_ALL=C.UTF-8 \
    --publish=7474:7474 --publish=7687:7687 \
    --volume=/home/your_user/neo4j_data:/data \
    --volume=/home/your_user/import:/import \
    neo4j:4.4.27 \
neo4j-admin import --verbose --nodes=Movie=/import/movie.csv \
--nodes=Actor=/import/person.csv \
--nodes=Genre=/import/genre.csv \
--relationships=ACTED_IN=/import/person_to_movie.csv \
--relationships=MOVIE_TYPE=/import/movie_to_genre.csv

### Expected behavior
import done by use docker image neo4j/neo4j:4.4.27 or neo4j/neo4j:4.4.29. or neo4j/neo4j:4.4.26

### Actual behavior
import done by use docker image  neo4j/neo4j:4.4.26
BUT by use docker image neo4j/neo4j:4.4.27
it shown 
Neo4j version: 4.4.27
Importing the contents of these files into /var/lib/neo4j/data/databases/neo4j:
Nodes:
  [Movie]:
  /import/movie.csv

  [Actor]:
  /import/person.csv

  [Genre]:
  /import/genre.csv

Relationships:
  ACTED_IN:
  /import/person_to_movie.csv

  MOVIE_TYPE:
  /import/movie_to_genre.csv

Available resources:
  Total machine memory: 503.8GiB
  Free machine memory: 7.893GiB
  Max heap memory : 29.97GiB
  Processors: 56
  Configured max memory: 426.4GiB
  High-IO: true

Nodes, started 2024-01-07 16:48:27.381+0000
[*Nodes:?? 1.004GiB---------------------------------------------------------------------------]    0 ∆    0
Done in 454ms
Prepare node index, started 2024-01-07 16:48:27.848+0000
[*DETECT:1.004GiB-----------------------------------------------------------------------------]10.0K ∆10.0K
Done in 112ms
Relationships, started 2024-01-07 16:48:27.967+0000
[*Relationships:?? 1.004GiB-------------------------------------------------------------------]20.0K ∆20.0K
Done in 162ms
Node Degrees, started 2024-01-07 16:48:28.161+0000
[>:?|*>--------------------------------------------------------|CALCULATE:1020MiB-------------] 120K ∆ 120K
Done in 81ms
Relationship --> Relationship 1-2/2, started 2024-01-07 16:48:28.289+0000
[>---------------------------|*LINK-----------------------------------|v:??-------------------] 120K ∆ 120K
Done in 122ms
RelationshipGroup 1-2/2, started 2024-01-07 16:48:28.423+0000
[*>:??---------------------------------------------------------------------|v:??--------------] 163K ∆ 163K
Done in 51ms
Node --> Relationship, started 2024-01-07 16:48:28.483+0000
[>:??----|*>-------------------------------------------|LINK--------|v:??---------------------] 273K ∆ 273K
Done in 104ms
Relationship <-- Relationship 1-2/2, started 2024-01-07 16:48:28.606+0000
[>-------------------------------|*LINK-------------------------------|v:??-------------------] 120K ∆ 120K
Done in 72ms
Count groups, started 2024-01-07 16:48:28.707+0000
[*>------------------------------------------------------------------------------|COUNT:1020Mi]  327 ∆  327
Done in 10ms
Gather, started 2024-01-07 16:48:28.737+0000
[*CACHE:5.261GiB------------------------------------------------------------------------------]  327 ∆  327
Done in 2s 597ms
Write, started 2024-01-07 16:48:31.347+0000
[*>:??--------------------------------------------|ENCODE--------------|v:??------------------]  327 ∆  327
Done in 11ms
Node --> Group, started 2024-01-07 16:48:31.373+0000
[>|*>------------------------------------|FIRST------------------------|v:??------------------]  546 ∆  546
Done in 10ms
Node counts and label index build, started 2024-01-07 16:48:31.903+0000
[>--------------|*LABEL INDEX----------------------------------------------|COUNT:1020MiB-----]10.0K ∆10.0K
Done in 40ms
Relationship counts and relationship type index build, started 2024-01-07 16:48:32.006+0000
[>-------------------|*RELATIONSHIP TYPE INDEX-------------------------|COUNT-----------------]30.0K ∆30.0K
Done in 61ms

IMPORT DONE in 5s 62ms. 
Imported:
  5044 nodes
  22351 relationships
  48016 properties
Peak memory usage: 1.032GiB
Import error: Temporary store is dislocated. It should be located under current database directory but instead located in: /data/databases/neo4j
Caused by:Temporary store is dislocated. It should be located under current database directory but instead located in: /data/databases/neo4j
java.lang.IllegalStateException: Temporary store is dislocated. It should be located under current database directory but instead located in: /data/databases/neo4j
	at org.neo4j.internal.batchimport.store.BatchingNeoStores.cleanup(BatchingNeoStores.java:511)
	at org.neo4j.internal.batchimport.store.BatchingNeoStores.close(BatchingNeoStores.java:493)
	at org.neo4j.internal.batchimport.ParallelBatchImporter.doImport(ParallelBatchImporter.java:111)
	at org.neo4j.importer.CsvImporter.doImport(CsvImporter.java:207)
	at org.neo4j.importer.CsvImporter.doImport(CsvImporter.java:172)
	at org.neo4j.importer.ImportCommand.execute(ImportCommand.java:277)
	at org.neo4j.cli.AbstractCommand.call(AbstractCommand.java:71)
	at org.neo4j.cli.AbstractCommand.call(AbstractCommand.java:34)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.neo4j.cli.AdminTool.execute(AdminTool.java:93)
	at org.neo4j.cli.AdminTool.main(AdminTool.java:79)

WARNING Import failed. The store files in /var/lib/neo4j/data/databases/neo4j are left as they are, although they are likely in an unusable state. Starting a database on these store files will likely fail or observe inconsistent records so start at your own risk or delete the store manually
java.lang.IllegalStateException: Temporary store is dislocated. It should be located under current database directory but instead located in: /data/databases/neo4j
	at org.neo4j.internal.batchimport.store.BatchingNeoStores.cleanup(BatchingNeoStores.java:511)
	at org.neo4j.internal.batchimport.store.BatchingNeoStores.close(BatchingNeoStores.java:493)
	at org.neo4j.internal.batchimport.ParallelBatchImporter.doImport(ParallelBatchImporter.java:111)
	at org.neo4j.importer.CsvImporter.doImport(CsvImporter.java:207)
	at org.neo4j.importer.CsvImporter.doImport(CsvImporter.java:172)
	at org.neo4j.importer.ImportCommand.execute(ImportCommand.java:277)
	at org.neo4j.cli.AbstractCommand.call(AbstractCommand.java:71)
	at org.neo4j.cli.AbstractCommand.call(AbstractCommand.java:34)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at org.neo4j.cli.AdminTool.execute(AdminTool.java:93)
	at org.neo4j.cli.AdminTool.main(AdminTool.java:79)

**Comment by aicodex** [comment]
Maybe because of the 4.4.27 fix:
"Fixes an issue where neo4j-admin import could run on an online database. This could result in not seeing the imported data until after a restart."?

**Comment by mnssn** [comment]
Thank you reporting the issue! Also thank you for the detailed bug report. :slightly_smiling_face:  You were right in that the issue was caused by the referenced 4.4.27 fix. 

An unintended side-effect of the fix caused two file paths to be differently resolved. Due to the layout of the docker container, the issue surfaced only in this environment, and not in standalone versions of the DBMS.

We have included a correction for this issue in the upcoming 4.4.31 release.
